projects/dotcms-js/src/lib/core/core-web.service.ts

```diff
@@ -135,77 +149,104 @@ export class CoreWebService {
      */
     public requestView(options: RequestOptionsArgs): Observable<ResponseView> {
         const request = this.getRequestOpts(options);
-
-        return this._http.request(request).pipe(
-            map((resp) => {
-                if (resp.json().errors && resp.json().errors.length > 0) {
+        return this.http.request(request).pipe(
+            filter(<T>(event: HttpEvent<T>) => event.type === HttpEventType.Response),
+            map((resp: HttpResponse<any>) => {
+                if (resp.body && resp.body.errors && resp.body.errors.length > 0) {
                     return this.handleRequestViewErrors(resp);
                 } else {
                     return new ResponseView(resp);
                 }
             }),
-            catchError((err: Response) => throwError(this.handleRequestViewErrors(err)))
+            catchError(<T>(err: HttpResponse<T>) => {
+                return throwError(this.handleRequestViewErrors(err));
+            })
         );
     }
 
-    private handleRequestViewErrors(resp: Response): ResponseView {
+    public subscribeTo(httpErrorCode: number): Observable<any> {
+        if (!this.httpErrosSubjects[httpErrorCode]) {
+            this.httpErrosSubjects[httpErrorCode] = new Subject();
+        }
+
+        return this.httpErrosSubjects[httpErrorCode].asObservable();
+    }
+
+    private handleRequestViewErrors<T>(resp: HttpResponse<T>): ResponseView {
         if (resp.status === 401) {
             this.router.navigate(['/public/login']);
-          }
+        }
 
         return new ResponseView(resp);
     }
 
-    public subscribeTo(httpErrorCode: number): Observable<any> {
-        if (!this.httpErrosSubjects[httpErrorCode]) {
-            this.httpErrosSubjects[httpErrorCode] = new Subject();
+    private handleHttpError(response: HttpErrorResponse): void {
+        if (!this.httpErrosSubjects[response.status]) {
+            this.httpErrosSubjects[response.status] = new Subject();
         }
 
-        return this.httpErrosSubjects[httpErrorCode].asObservable();
+        this.httpErrosSubjects[response.status].next(response);
     }
 
-    private getRequestOpts(options: RequestOptionsArgs): Request {
-        const headers: Headers = this._apiRoot.getDefaultRequestHeaders();
+    private getRequestOpts<T>(options: RequestOptionsArgs): HttpRequest<T> {
```

Comment:
You are passing the generic here but where you call this function you are not passing any generics.